### PR TITLE
Housekeeping: Update dependencies - Nov 25'

### PR DESCRIPTION
**Pull request by KCeh**
## Summary

Just housekeeping, updating project and dependencies 

## Changes

- gitignore update
- dependecies update

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [x] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,6 +49,7 @@ captures/
 .idea/modules.xml
 # Comment next line if keeping position of elements in Navigation Editor is relevant for you
 .idea/navEditor.xml
+.idea/*
```
Comment (AsimRibo) [inline .gitignore]: Since we already have specific **.idea** patterns, what is the idea behind wildcard? Are specific idea rules even needed then?

**Review by AsimRibo — APPROVED** [review]
Image
Lookin good.

File: .gitignore (hunk 1 above)
Comment (KCeh) [reply .gitignore]: I just added it on top of already existing ignore rules, specific rules are not needed then, but I don't think it is a big problem

Comment (JonatanPlesko) [reply .gitignore]: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems?page=4

Ignoring the entire .idea folder is a bit of a nuclear option and not recommended by Jetbrains

Comment (KCeh) [reply .gitignore]: Well I guess then we will improve the rules

Comment (JonatanPlesko) [reply .gitignore]: I would consider doing a bit of a hybrid approach as outlined in the article:
https://blog.sebastiano.dev/anatomy-of-a-gitignore/

Since I noticed some plugins like to put stuff in `.idea` that should _not_ be shared, so I do think it's more practical to ignore _by default_, but intentionally un-ignore some stuff that should be shared.

Comment (JonatanPlesko) [reply .gitignore]: @antunflas @schalke95 thoughts?

Comment (KCeh) [reply .gitignore]: What do you think about [this config](https://gist.github.com/rock3r/ac33be8ae9df940e254ca4b4bb06fa9d?ref=blog.sebastiano.dev)?

related IDE config:
```
*.iml
*.ipr
*.iws
/.idea/*

# Exclude non-user-specific stuff
!.idea/.name
!.idea/codeInsightSettings.xml
!.idea/codeStyles/
!.idea/copyright/
!.idea/dataSources.xml
!.idea/detekt.xml
!.idea/encodings.xml
!.idea/externalDependencies.xml
!.idea/file.template.settings.xml
!.idea/fileTemplates/
!.idea/icon.svg
!.idea/inspectionProfiles/
!.idea/runConfigurations/
!.idea/scopes/
!.idea/vcs.xml
```

These are files I am getting as diff before nuclear option of ignoring whole .idea folder:

Screenshot 2025-11-28 at 13 30 36


Looking at what they do, it might be useful to keep (but nothing super important):
```
.name
runConfigurations.xml
kotlinc.xml
AndroidProjectSystem.xml

```

Comment (JonatanPlesko) [reply .gitignore]: I think some of those exclusions might be outdated, such as the `runConfiguration` folder, which I don't see nowadays and instead only see the .xml file 🤔 same with `detekt.xml`, we keep detekt rules elsewhere. But I do agree with your list of exclusions!
